lib/features/children/add_member/widgets/allowance_counter.dart

```diff
@@ -10,12 +10,14 @@ class AllowanceCounter extends StatefulWidget {
     required this.currency,
     this.initialAllowance,
     this.onAllowanceChanged,
+    this.isAmountNullable = false,
     super.key,
   });
 
   final String? currency;
   final int? initialAllowance;
   final void Function(int allowance)? onAllowanceChanged;
+  final bool isAmountNullable;
```

Comment:
For clarity I would rename this to canAmountBeZero (as nullable is something else for me)